src/pivot_cache.jl

```diff
@@ -1,29 +1,31 @@
 """
-x = pivot_cache(A, b; ...)
+    x = pivot_cache(A, b; ...)
 
-Solves non-negative least-squares problem by block principal pivoting method
+Solve non-negative least-squares problem by block principal pivoting method
 (Algorithm 1) described in Kim & Park (2011).
 
 Optional arguments:
-    tol: tolerance for nonnegativity constraints
-    max_iter: maximum number of iterations
+* `tol` tolerance for nonnegativity constraints
+ default `10^floor(log10(eps(T)^0.5))` which is `1e-8` for `Float64`
+* `max_iter` maximum number of iterations, default `30 * size(A,2)`
 
 References:
-	J. Kim and H. Park, Fast nonnegative matrix factorization: An
-	active-set-like method and comparisons, SIAM J. Sci. Comput., 33 (2011),
-	pp. 3261–3281.
+    J. Kim and H. Park, Fast nonnegative matrix factorization: An
+    active-set-like method and comparisons, SIAM J. Sci. Comput., 33 (2011),
+    pp. 3261–3281.
 """
-function pivot_cache(AtA,
-                    Atb::AbstractVector{T};
-                    tol::Float64=1e-8,
-                    max_iter=30*size(AtA,2)) where {T}
-
+function pivot_cache(
+    AtA,
+    Atb::AbstractVector{T};
+    tol::Real = 10^floor(log10(eps(T)^0.5)),
+    max_iter=30 * size(AtA,2),
+) where {T <: AbstractFloat}
```

Comment:
I don't think we need to restrict `T` here and should remain as generic as possible ( ForwardDiff.jl, Unitful.jl ect... ) as long as there is no strict need to do so. Could actually be a breaking change.

Reply:
The problem is that `eps` is defined only for `AbstractFloat` types.
Not even for Complex float types:
https://discourse.julialang.org/t/eps-epsilon-for-complex-floats/59603
But I will find another way, like using the original default 1e-8 for any non-AbstractFloat type.
That will address the original issue while also maintaining generality.